bin/node/src/main.rs

```diff
@@ -9,7 +9,20 @@ use sc_network::config::Role;
 use sc_service::PartialComponents;
 
 fn main() -> sc_cli::Result<()> {
-    let cli = Cli::parse();
+    let mut cli = Cli::parse();
+
+    match cli.run.import_params.pruning_params.pruning.as_ref() {
+        Some(mode) => {
+            if mode != "archive" {
+                println!("Pruning not supported. Switching to 'archive' mode.");
```

Comment:
It's better to log it I think -- `info!` probably.

Reply:
Logger is not initialized here yet :( Actually, where is it initialized @h4nsu? Haven't found that. Does it happen inside Substrate?

Reply:
It's not possible to use the logger here, it gets initialized later, in differerent `match` branches separately.

Reply:
@maciejnems I think somewhere in `cli.create_runner`

Reply:
[client/cli/src/config.rs](https://github.com/Cardinal-Cryptography/substrate/blob/5f0aa1feb7250ac7b8c1b9928f87b2420b530e22/client/cli/src/config.rs#L651)